ui-spacetimechart/src/components/workSchedules/WorkSchedules.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+import Region from './Region';
+import { type WorkSchedule } from '../../types';
+
+type WorkSchedulesProps = {
+  workSchedules: WorkSchedule[];
+  pattern: string;
+};
+/**
+ * Display the workschedule projection on the Space time chart.
+ * a workschedule can contains several track occupied during a period of this
+ * each is represented by a rectangle region
+ */
+function WorkSchedules({ workSchedules, pattern }: WorkSchedulesProps) {
+  return workSchedules.flatMap((ws) =>
+    ws.intersections.map(([spaceStart, spaceEnd]) => (
+      <Region
+        key={`${ws.timeStart}-${spaceStart}`}
```

Comment:
I'm worried we might end up with duplicate keys here.

Reply:
added some more stuff.

Reply:
I _think_ this is fine for the purposes of work schedules, but if we're going to expose `PatternRect` as a generic layer in the future I don't think this is something we can assume. We wouldn't need keys if the layer was responsible for painting multiple rects instead of a single one.